kani-driver/src/cbmc_output_parser.rs

```diff
@@ -0,0 +1,991 @@
+
+/// Enum that represents a parser item.
+/// See the parser for more information on how they are processed.
+#[derive(Debug, Deserialize)]
+#[serde(untagged)]
+enum ParserItem {
+    Program {
+        program: String,
+    },
+    #[serde(rename_all = "camelCase")]
+    Message {
+        message_text: String,
+        message_type: String,
+    },
+    Result {
+        result: Vec<Property>,
+    },
+    #[serde(rename_all = "camelCase")]
+    ProverStatus {
+        _c_prover_status: String,
+    },
+}
+
+impl ParserItem {
+    /// Determines if an item must be skipped or not.
+    fn must_be_skipped(&self) -> bool {
+        matches!(&self, ParserItem::Message { message_text, .. } if message_text.starts_with("Building error trace") || message_text.starts_with("VERIFICATION"))
+    }
+}
+
+/// Struct that represents a property.
+///
+/// Note: `reach` is not part of the parsed data, but it's useful to annotate
+/// its reachability status.
+#[derive(Clone, Debug, Deserialize)]
+pub struct Property {
+    pub description: String,
+    pub property: String,
+    #[serde(rename = "sourceLocation")]
+    pub source_location: SourceLocation,
+    pub status: CheckStatus,
+    pub reach: Option<CheckStatus>,
+    pub trace: Option<Vec<TraceItem>>,
+}
+
+/// Struct that represents a source location.
+///
+/// Source locations may be completely empty, which is why
+/// all members are optional.
+#[derive(Clone, Debug, Deserialize)]
+pub struct SourceLocation {
+    pub column: Option<String>,
+    pub file: Option<String>,
+    pub function: Option<String>,
+    pub line: Option<String>,
+}
+
+impl SourceLocation {
+    /// Determines if fundamental parts of a source location are missing.
+    fn is_missing(&self) -> bool {
+        self.file.is_none() && self.function.is_none()
+    }
+}
+
+/// `Display` implement for `SourceLocation`.
+///
+/// This is used to format source locations for individual checks. But source
+/// locations may be printed in a different way in other places (e.g., in the
+/// "Failed Checks" summary at the end).
+///
+/// Source locations formatted this way will look like:
+/// `<file>:<line>:<column> in function <function>`
+/// if all attributes were specified. Otherwise, we:
+///  * Omit `in function <function>` if the function isn't specified.
+///  * Use `Unknown file` instead of `<file>:<line>:<column>` if the file isn't
+///    specified.
+///  * Lines and columns are only formatted if they were specified and preceding
+///    attribute was formatted.
+impl std::fmt::Display for SourceLocation {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let mut fmt_str = String::new();
+        if let Some(file) = self.file.clone() {
+            let file_str = filepath(file);
+            fmt_str.push_str(file_str.as_str());
+            if let Some(line) = self.line.clone() {
+                let line_str = format!(":{}", line);
+                fmt_str.push_str(line_str.as_str());
```

Comment:
I think it would be cleaner to use `write!(&mut fmt_str, ":{line}")?` instead of these two lines. Similar in the rest of the function (and maybe other instances in the file.